votor/src/certificate_pool.rs

```diff
@@ -698,15 +700,35 @@ impl CertificatePool {
     }
 
     pub fn get_certs_for_standstill(&self) -> Vec<Arc<CertificateMessage>> {
+        let (highest_finalized_with_notarize_slot, has_fast_finalize) = self
+            .highest_finalized_with_notarize
+            .unwrap_or((self.root, false));
         self.completed_certificates
             .iter()
             .filter_map(|(cert_id, cert)| {
-                if Some(cert.certificate.slot()) >= self.highest_finalized_with_notarize {
+                let cert_to_send = match cert
```

Comment:
I think it would be cleaner if we avoided the nesting here with a single match statement, also seems easier to use `cert_id` instead of `cert` indirection
```rust
match (cert_id.slot().cmp(&highest_finalized_with_notarize_slot), cert_id.certificate_type, has_fast_finalize) => {
  (Ordering::Greater, _, _) 
  | (Ordering::Equal, Finalize | Notarize, false),
  | (Ordering::Equal, FastFinalize, true) => Some(cert.clone()),
  (Ordering::Equal, FastFinalize, false) => panic!("Programmer Error"),
  _ => None
}
```
Something like this?

Reply:
Done.

I wouldn't panic here, we can totally get FastFinalize cert after highest_finalized_with_notarize_slot is set.

Reply:
hmm but in that case we update the bool right?
```
 if self
                    .highest_finalized_with_notarize
                    .map_or(true, |(s, _)| s <= slot)
                {
                    self.highest_finalized_with_notarize = Some((slot, true));
                }
```
since you use a "<=" here?

Reply:
It only works when this whole thing is still single threaded. Sure.